bolt-modules/boltlib/lib/puppet/functions/resolve_references.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+# Evaulates all _plugin references in a hash and returns the resolved reference data.
+Puppet::Functions.create_function(:resolve_references) do
+  # Resolve references.
+  # @param references A hash of reference data to resolve.
+  # @return A hash of resolved reference data.
+  # @example Resolve a hash of reference data
+  #   $references = {
+  #     "targets" => [
+  #       "_plugin" => "terraform",
+  #       "dir" => "path/to/terraform/project",
+  #       "resource_type" => "aws_instance.web",
+  #       "uri" => "public_ip"
+  #     ]
+  #   }
+  #
+  #   resolve_references($references)
+  dispatch :resolve_references do
+    param 'Hash[String[1], Any]', :references
```

Comment:
I think the param and return types should both be `Data`. `Data` means the subset of Puppet data types which are legal JSON (and therefore what we can pass to and receive from a task/plugin).

Reply:
Updated!